cmd/juju/secretbackends/modelsecretbackend.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package secretbackends
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/juju/cmd/v4"
+	"github.com/juju/errors"
+
+	api "github.com/juju/juju/api/client/modelconfig"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
+	secretbackenderrors "github.com/juju/juju/domain/secretbackend/errors"
+	"github.com/juju/juju/internal/secrets/provider"
+)
+
+type modelSecretBackendCommand struct {
+	modelcmd.ModelCommandBase
+
+	getAPIFunc func() (ModelSecretBackendAPI, error)
+
+	secretBackendName *string
+}
+
+// ModelSecretBackendAPI is the mdoel secret backend client API.
+type ModelSecretBackendAPI interface {
+	GetModelSecretBackend(ctx context.Context) (string, error)
+	SetModelSecretBackend(ctx context.Context, secretBackendName string) error
+	Close() error
+}
+
+// NewModelSecretBackendCommand returns a command to get or set secret backend config for the current model.
+func NewModelSecretBackendCommand() cmd.Command {
+	c := &modelSecretBackendCommand{}
+	c.getAPIFunc = c.secretBackendAPI
+	return modelcmd.Wrap(c)
+}
+
+func (c *modelSecretBackendCommand) secretBackendAPI() (ModelSecretBackendAPI, error) {
+	root, err := c.NewAPIRoot()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return api.NewClient(root), nil
+}
+
+const (
+	modelSecretBackendDoc = `
+Sets or displays the secret backend for the current model.
+`
+	modelSecretBackendExamples = `
+Display the secret backend for the current model:
+
+   juju model-secret-backend
+
+Set the secret backend to myVault for the current model:
+
+   juju model-secret-backend myVault
+`
+)
+
+// Info implements cmd.Command.
+func (c *modelSecretBackendCommand) Info() *cmd.Info {
+	return jujucmd.Info(&cmd.Info{
+		Name:     "model-secret-backend",
```

Comment:
Do we think we have got this name correct?

If I run `juju help` and look at all of the commands we have very very few reference the word model even though they are scoped to the model the user is working on. I am just trying to understand and make sure this is in the spirit of what we have now.

Apologies as I am not trying to stop this PR I am just trying to ask the question is this correct and in the same spirit as the rest of our commands?

Reply:
This can be ignored. We have talked about current expectations and agreed this is correct.